core/iwasm/fast-jit/fe/jit_emit_control.c

```diff
@@ -595,12 +595,17 @@ handle_next_reachable_block(JitCompContext *cc, uint8 **p_frame_ip)
         }
         else if (block->incoming_insns_for_end_bb) {
             *p_frame_ip = block->wasm_code_end + 1;
-            return handle_op_end(cc, p_frame_ip, false);
+            return handle_op_end(cc, p_frame_ip, block == first_block);
```

Comment:
Why need to add `block == first_block`? For case below:
```wast
(module
  (type $0 (func (result i32)))
  (export "test" (func $1))
  (func $1 (type $0)
    (result i32)
    i32.const 5678
    (block $l (param i32) (result i32)
      i32.const 1234
      i32.add
      i32.const 1
      br 0
      drop
      i32.const 666
    )
  )
)
```
Dumplicated `JMP` instruction are generated in L2:
```
L2:
    ; PREDS(L0)
    ; BEGIN_BCIP=0x0027 END_BCIP=0x0038
    ADD             i7, 0x162e, 0x4d2
    ADD             I21, I0, 0x90L
    STI32           0x1, I21, 0x0
    JMP             L3
    JMP             L3
    ; SUCCS(L3)
```

Reply:
try to handle one of the above problems. (else return end)

```
  (func (export "e0") (param i32) (result i32)
    (i32.const 3)
    (if (param i32) (result i32) (local.get 0)
      (then
        (block (param i32) (result i32)
          (drop)
          (i32.const 7)
          (return)
        )
      )
      (else
        (return)
      )
    )
    (return)
  )
```
And missing  `jmp` in `ELSE` branch

```
L4:
    ; PREDS(L2)
    ; BEGIN_BCIP=0x002f END_BCIP=0x0030
    LDI32           i9, I0, 0x94        ;-> i32.const 3
    LDI64           I23, I0, 0x0        ;-> function return
    LDI64           I24, I23, 0x70
    STI32           i9, I24, 0x0
    ADD             I24, I24, 0x4L
    STI64           I24, I23, 0x70
    STI64           I0, I15, 0xc90
    STI64           I23, I15, 0x68
    MOV             I0, I23
    RETURNBC        0x0, VOID, VOID
    STI32           i9, I0, 0x94        ;-> gen_commit_value()
                                                 ;-> miss a JMP ?
    ; SUCCS()
```

Guess we need more condition

Reply:
No need to jmp here，RETURNBC will return to caller, the following insns won't be executed. In fact, the STI32 should not be generated. Guess there are unneeded commit_values called.